torch/csrc/lazy/core/ir.h

```diff
@@ -185,7 +205,8 @@ class TORCH_API Node {
   // The hash value of this node.
   hash_t node_hash_;
   // The hash value of the graph rooted at this node.
```

Comment:
please update this comment, put the explanation from the PR summary, why you use without_sizes for graph cache and with_sizes for shape cache

Reply:
> looks good to me. One scenario I was wondering about is what happens when `dynamic_shapes` changes in between iterations (from dynamic to static or vice versa). Do we need add this boolean to the overall graph hash or just hashing not hashing full shapes should be enough?

Yep, I think if we want to guard that scenario, we can add the dynamic_shapes knob to the graph hash. It may not be good to add this knob in Shape::hash since this way, the knob will be hashed for each Shape. We may find some places to hash the knob only once for the entire graph